pkg/storage/backend/s3/config.go

```diff
@@ -2,17 +2,38 @@ package s3
 
 import (
 	"flag"
+	"net/http"
+	"time"
 
 	"github.com/cortexproject/cortex/pkg/util/flagext"
 )
 
+// HTTPConfig stores the http.Transport configuration for the s3 minio client.
+type HTTPConfig struct {
+	IdleConnTimeout       time.Duration `yaml:"idle_conn_timeout"`
+	ResponseHeaderTimeout time.Duration `yaml:"response_header_timeout"`
+	InsecureSkipVerify    bool          `yaml:"insecure_skip_verify"`
+
+	// Allow upstream callers to inject a round tripper
+	Transport http.RoundTripper `yaml:"-"`
+}
+
+// RegisterFlagsWithPrefix registers the flags for TSDB s3 storage with the provided prefix
+func (cfg *HTTPConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.DurationVar(&cfg.IdleConnTimeout, prefix+"s3.http.idle-conn-timeout", 0, "If set, the time an idle connection will remain idle before closing.")
```

Comment:
Should we use default values from vendor/github.com/thanos-io/thanos/pkg/objstore/s3/s3.go:49 (90s idle conns timeout, and 2min response header timeout) instead?

Reply:
Yes, I think that makes sense.

Reply:
I think that will be a change from what Cortex does now, so let's document that in the CHANGELOG as well.